frontend/public/components/routes.tsx

```diff
@@ -200,26 +203,49 @@ const RouteListRow: React.SFC<RoutesRowProps> = ({obj: route}) => <ResourceRow o
   <div className="col-lg-2 hidden-md hidden-sm hidden-xs"><RouteStatus obj={route} /></div>
 </ResourceRow>;
 
-const TLSSettings = props => <span>
-  {!props.tls && 'TLS is not enabled.'}
-  {props.tls && <dl>
-    <dt>Termination Type</dt>
-    <dd>{props.tls.termination}</dd>
-    <dt>Insecure Traffic</dt>
-    <dd>{props.tls.insecureEdgeTerminationPolicy || '-'}</dd>
-    <dt>Certificate</dt>
-    <dd>{props.tls.certificate || '-'}</dd>
-    <dt>Key</dt>
-    <dd>{props.tls.key || '-'}</dd>
-    <dt>CA Certificate</dt>
-    <dd>{props.tls.caCertificate || '-'}</dd>
-    {_.get(props.tls, 'termination') === 'reencrypt' && <span>
-      <dt>Destination CA Cert</dt>
-      <dd>{props.tls.destinationCACertificate || '-'}</dd>
-    </span>}
-  </dl>
+class TLSSettings extends SafetyFirst<TLSDataProps, TLSDataState> {
+  constructor(props) {
+    super(props);
+    this.state = { showPrivateKey: false };
+    this.toggleKey = this.toggleKey.bind(this);
+  }
+
+  toggleKey() {
+    this.setState({ showPrivateKey: !this.state.showPrivateKey });
+  }
+
+  render() {
+    const { showPrivateKey } = this.state;
+    const { tls } = this.props;
+    const visibleKeyValue = showPrivateKey ? tls.key : <MaskedData /> ;
+
+    return !tls ? 'TLS is not enabled.'
+      : <dl>
+        <dt>Termination Type</dt>
+        <dd>{tls.termination}</dd>
+        <dt>Insecure Traffic</dt>
+        <dd>{tls.insecureEdgeTerminationPolicy || '-'}</dd>
+        <dt>Certificate</dt>
+        <dd>{tls.certificate ? <CopyToClipboard value={tls.certificate}/> : '-'}</dd>
+        <dt className="co-m-route-tls-reveal__title">Key {tls.key &&
+          <button className="btn btn-link co-m-route-tls-reveal__btn" type="button" onClick={this.toggleKey}>
+            {
+              showPrivateKey
+                ? <React.Fragment><i className="fa fa-eye-slash" aria-hidden="true"></i> Hide</React.Fragment>
+                : <React.Fragment><i className="fa fa-eye" aria-hidden="true"></i> Reveal</React.Fragment>
+            }
+          </button>}
+        </dt>
+        <dd>{tls.key ? <CopyToClipboard value={tls.key} visibleValue={visibleKeyValue}/> : '-'}</dd>
+        <dt>CA Certificate</dt>
+        <dd>{tls.caCertificate ? <CopyToClipboard value={tls.caCertificate}/> : '-'}</dd>
+        {tls.termination === 'reencrypt' && <React.Fragment>
+          <dt>Destination CA Cert</dt>
+          <dd>{tls.destinationCACertificate ? <CopyToClipboard value={tls.destinationCACertificate}/> : '-'}</dd>
+        </React.Fragment>}
+      </dl>;
   }
-</span>;
+}
 
```

Comment:
Missed on the prev PR for this, but you might consider upgrading your `_.each` to a `_.reduce` & eliminate some bookkeeping in `calcTrafficPercentage`:

```JavaScript 
const calcTrafficPercentage = (weight: number, route: any) => {
  return _.reduce(route.spec.alternateBackends, (result, alternate) => {
    return result += alternate.weight;
  }, 0 + route.spec.to.weight);
};
```

Reply:
Hi @benjaminapetersen, still trying to understand this formula.  When I plug it in, all percentages come back as 100%:
![image](https://user-images.githubusercontent.com/12733153/41046792-06b59564-6979-11e8-983a-871c527a7f8a.png)
Whereas, they should equal the Origin Console Route's Traffic table:
![image](https://user-images.githubusercontent.com/12733153/41046824-1cbaf1ba-6979-11e8-8461-ee880d2a1689.png)

Reply:
Got it:
![image](https://user-images.githubusercontent.com/12733153/41047234-161baa56-697a-11e8-9102-f4bc5dabd298.png)

```
let totalWeight = _.reduce(route.spec.alternateBackends, (result, alternate) => {
    return result += alternate.weight;
  }, 0 + route.spec.to.weight);

  return (weight/totalWeight*100).toFixed(1);
```

Reply:
Sorry for oops on syntax, didn't pull your code & test, but was just pointing out an alt to `.each` to abstract some of bookkeeping.